lib/elixir/lib/calendar/duration.ex

```diff
@@ -44,6 +44,8 @@ defmodule Duration do
           | {:second, integer}
           | {:microsecond, {integer, 0..6}}
 
+  @type duration :: t | [unit_pair]
```

Comment:
We should add some docs. Basically that says we specify the duration either as a data type or as a keyword. :)

Reply:
Not sure if the current formulation is more helpful than mentioning that `t:Duration.duration/0` is a union of `t:Duration.t/0` and a keyword list of `t:Duration.unit_pair/0`. Happy give it another go if you think https://github.com/elixir-lang/elixir/pull/13476/commits/0cefcd17fd27c32b5e364f41b2f335c5bdd7e28a is not clear enough.